src/Microsoft.PowerApps.TestEngine/PowerApps/PowerAppFunctions.cs

```diff
@@ -143,8 +134,29 @@ private async Task<bool> CheckIfAppIsIdleAsync()
             return controlDictionary;
         }
 
+        private async void checkAndHandleIfLegacyPlayer()
+        {
+            try
+            {
+                var checkJSSDKExistExpression = "typeof PowerAppsTestEngine";
+                var result = await _testInfraFunctions.RunJavascriptAsync<string>(checkJSSDKExistExpression);
+                if (result.ToLower() == "undefined")
+                {
+                    _singleTestInstanceState.GetLogger().LogTrace("Legacy WebPlayer in use, injecting embedded JS");
+                    await _testInfraFunctions.AddScriptTagAsync(GetFilePath(Path.Combine("JS", "CanvasAppSdk.js")), null);
+                    await _testInfraFunctions.AddScriptTagAsync(GetFilePath(Path.Combine("JS", "PublishedAppTesting.js")), PublishedAppIframeName);
+                }
+            }
+            catch (Exception ex)
+            {
+                _singleTestInstanceState.GetLogger().LogDebug(ex.ToString());
+                return false;
```

Comment:
shouldn't be returning anything in a void function

Reply:
Copy paste error :) Updated